src/cls/rgw/cls_rgw_types.h

```diff
@@ -130,6 +132,8 @@ struct rgw_bucket_dir_entry_meta {
       accounted_size = size;
     if (struct_v >= 5)
       ::decode(user_data, bl);
+    if (struct_v >= 6)
```

Comment:
you need to set accounted_entry as true for older version which don't have this flag.
otherwise when you delete them you won't remove them from the accounting (which there were already accounted for)